todoApp/src/App.tsx

```diff
@@ -1,15 +1,51 @@
-import React from "react";
+import React, { useState } from "react";
 import styles from "./App.module.css";
 import Button from "./components/Button";
-import Date from "./components/Date";
-import TodoLists from "./components/TodoLists";
+import DateContainer from "./components/DateContainer";
+import DisplayTodo from "./components/DisplayTodo";
+
+export interface TodoListInterface {
+  id: string;
+  title: string;
+  date: string;
+  isCompleted: boolean;
+}
 
 const App = () => {
+  let [todoTasks, setTodoTasks] = useState<TodoListInterface[]>(
+    JSON.parse(localStorage.getItem("todo") || "[]")
+  );
+
+  const buttonHandler = () => {
+    const todoArray: TodoListInterface[] = JSON.parse(
+      localStorage.getItem("todo") || "[]"
+    );
+    setTodoTasks(todoArray);
+  };
+
+  const curTime = new Date();
+  todoTasks = todoTasks.filter((todo: TodoListInterface) => {
+    let curDate = `${curTime.getDate()}/${
+      curTime.getMonth() + 1
+    }/${curTime.getFullYear()}`;
+    return curDate === todo.date;
+  });
+  localStorage.setItem("todo", JSON.stringify(todoTasks));
```

Comment:
side effects should be inside a hook rather than component body itself.
It is the recommended way.